src/ui/public/vislib/__tests__/lib/vis_config.js

```diff
@@ -0,0 +1,113 @@
+import d3 from 'd3';
+import _ from 'lodash';
+import ngMock from 'ng_mock';
+import expect from 'expect.js';
+import VislibLibVisConfigProvider from 'ui/vislib/lib/vis_config';
+import PersistedStatePersistedStateProvider from 'ui/persisted_state/persisted_state';
+
+describe('Vislib VisConfig Class Test Suite', function () {
+  let visConfig;
+  let el;
+  const data = {
+    hits: 621,
+    ordered: {
+      date: true,
+      interval: 30000,
+      max: 1408734982458,
+      min: 1408734082458
+    },
+    series: [
+      {
+        label: 'Count',
+        values: [
+          {
+            x: 1408734060000,
+            y: 8
+          },
+          {
+            x: 1408734090000,
+            y: 23
+          },
+          {
+            x: 1408734120000,
+            y: 30
+          },
+          {
+            x: 1408734150000,
+            y: 28
+          },
+          {
+            x: 1408734180000,
+            y: 36
+          },
+          {
+            x: 1408734210000,
+            y: 30
+          },
+          {
+            x: 1408734240000,
+            y: 26
+          },
+          {
+            x: 1408734270000,
+            y: 22
+          },
+          {
+            x: 1408734300000,
+            y: 29
+          },
+          {
+            x: 1408734330000,
+            y: 24
+          }
+        ]
+      }
+    ],
+    xAxisLabel: 'Date Histogram',
+    yAxisLabel: 'Count'
+  };
+
+  beforeEach(ngMock.module('kibana'));
+  beforeEach(ngMock.inject(function (Private) {
+    const VisConfig = Private(VislibLibVisConfigProvider);
+    const PersistedState = Private(PersistedStatePersistedStateProvider);
+    el = d3.select('body')
+      .attr('class', 'vis-wrapper')
+      .node();
+
+    visConfig = new VisConfig({
+      type: 'histogram',
+      el: el
+    }, data, new PersistedState());
+  }));
+
+  describe('get Method', function () {
+    it('should be a function', function () {
+      expect(_.isFunction(visConfig.set)).to.be(true);
```

Comment:
Not a big deal, but I generally get a little alarm bell when I see assertions against booleans, because failing test messages tend to be less helpful, e.g. `expected true but got false`. Usually they can be structured in a way that will give more info, e.g.

```javascript
expect(typeof visConfig.set).to.be.('function');
```

Will give a message along the lines of `expected 'function' but got 'object'`. I'm OK with the way it is now though, I just wanted to throw the thought out there. Your call. :)